datadog/resource_datadog_powerpack.go

```diff
@@ -602,6 +616,51 @@ func buildDatadogPowerpack(ctx context.Context, d *schema.ResourceData) (*datado
 
 }
 
+func normalizeDashboardWidgetDef(widgetDef map[string]interface{}) (map[string]interface{}, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	// Dashboard widgets set live span at the widget level, we don't allow that for powerpack widgets
+	// where live span is set at the resource level.
+	if widgetDef["live_span"] != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  fmt.Sprintf("live_span must be set at the powerpack level and will be applied to all widgets"),
+		})
+		return nil, diags
+	}
+
+	if widgetDef["request"] != nil {
+		// Distribution/change/heatmap widgets have a "requests" field, while API Spec has a "request" field
+		// Here we set the "requests" field and remove "request"
+		widgetDefRequests := *widgetDef["request"].(*[]map[string]interface{})
```

Comment:
**question:** Is there any risk that this won't cast (i.e. should we check)?

Reply:
the cast itself won't fail, but we do need to check if request has length > 0. a definition like 
```hcl
  widget {
    change_definition {
      title = "Widget Title"
    }
  }
```
causes the provider to fail with no meaningful error message.
I think the provider is discarding the api error because of how verbose the api error is. As an example, 
```bash
curl -X POST "https://api.datadoghq.com/api/v2/powerpacks" \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "DD-API-KEY: ${DD_API_KEY}" \
-H "DD-APPLICATION-KEY: ${DD_APP_KEY}" \
-d @- << EOF
{
    "data": {
        "attributes": {
            "description": "Test Powerpack 2",
            "group_widget": {
                "definition": {
                    "layout_type": "ordered",
                    "title": "Sample Powerpack",
                    "type": "group",
                    "widgets": [
                        {
                            "definition": {
                                "requests": [],
                                "title": "Widget Title",
                                "type": "change"
                            }
                        }
                    ]
                }
            },
            "name": "Sample Powerpack"
        },
        "type": "powerpack"
    }
}
EOF
```
gives me an api error thats 1.2 megabytes, where the first sentence explains the problem `["Invalid group widget for powerpack. Error: Invalid inner widget definition at position 0 of type change. Error: [] is too short.` followed by 1.2 megabytes of schema-related junk. is there anything that can be done about this on the backend side?

Reply:
Good catch! While the cast won't fail, I did add a check for length of `request` - if it is of length 0, then an error will be thrown:
```
resource "datadog_powerpack" "foo2" {
  name = "Sample Powerpack"
  tags = ["tag:foo1", "tag:foo2"]
  description = "Test Powerpack 2"

  template_variables {
    defaults = ["defaults"]
    name     = "datacenter"
  }
  widget {
    change_definition {
      title = "Widget Title"
    }
  }
}
```
returns:
```
datadog_powerpack.foo2: Creating...
╷
│ Error: at least one request should be defined for widget: change
│ 
│   with datadog_powerpack.foo2,
│   on main.tf line 90, in resource "datadog_powerpack" "foo2":
│   90: resource "datadog_powerpack" "foo2" {
│ 
╵
```
This also works for widgets of type `query_value` and `toplist` which have a required `requests` field.